### DISCUSSION TITLE
New panel in charging stats: Charging Cost at SuC

### DISCUSSION THREAD

**Discussion by CrowMash** [teslamate-org/teslamate#2753] [General]
Hi!

First of all, thank you very much for the v1.27.0, lots of really good new features!

Regarding of the new panel "Charging Cost at SuC", I find it very useful but as described in the pull request #2448, "The data will depend on whether the location of the recharges is correct or not".

In my case, only 4 out of 9 SuC are correctly named...

Maybe if the request is slightly modified to add Geofences it could minimise the impact of wrong addresses?

```
SELECT
  sum(cp.cost)
FROM
  charging_processes cp 
LEFT JOIN
  addresses addr ON addr.id = address_id 
LEFT JOIN
  geofences geo ON geo.id = geofence_id 
WHERE
  $__timeFilter(end_date)
  AND (addr.name LIKE '%Supercharger%' OR geo.name LIKE '%Supercharger%')
  AND cp.cost IS NOT NULL
  AND cp.car_id = $car_id;
```

Again, thank you very much for the hard work.

**Comment by adriankumpf** [comment]
Implemented in e60ee3439d1d4e6d7eba810ce5c42966b2474f8c

**Reply by CrowMash** [reply]
Nice! Thanks!

**Reply by ejseguinte** [reply]
Thank you.

**Comment by fergymcferguson** [comment]
Sorry if this is a stupid question, but do I have to set geofences for superchargers for this to populate? Or should it automatically pull the data?  Also, I just updated to 1.27 - will it pull historical data or only log new data?

**Reply by cwanja** [reply]
It will calculate if the geotag address or a geofence has Suoercharger in the name. And it calculates all historical and future data, assuming it meets the above criteria. 

No stupid questions.

**Reply by fergymcferguson** [reply]
Thank you. Interesting. Here are a few examples from my charge history and they’re not populating. I’m not sure where to add the associated charge rates for those locations without creating a geofence assuming it’s supposed to pull that from the Tesla api. 
![2FFA9C29-1E2C-490B-8FB1-E917EE202DB8](https://user-images.githubusercontent.com/65431183/182977777-84bcdb99-7f48-4bc1-b68e-2a9225102499.jpeg)

**Reply by cwanja** [reply]
Click the `-` in the first column after the time charging. You can assign individual charge costs without geofences. 

Edit: I do not believe the API pulls costs.

**Reply by CrowMash** [reply]
The purpose of this panel is only to display the total cost of charges where the address name or the geofence name contains "supercharger" assuming you already have set charges cost in "Charges" dashboard. If you want to add rates, you have to create geofences, otherwise you'll have to set cost manually for each charges.

**Reply by fergymcferguson** [reply]
Ahhh got it. I was under the impression that it was going to automatically pull this data from the Tesla app. Thank you!  I’ve got it all sorted and updated with the costs now!

**Reply by prisnagka** [reply]
> It will calculate if the geotag address or a geofence has Suoercharger in the name. And it calculates all historical and future data, assuming it meets the above criteria.
> 
> No stupid questions.

Sadly, not all Tesla Superchargers have the name "Supercharger" in them :( Shocking, I know.

One example is the "Jämtkraft/Green Highway, Krokom", located in mid-Sweden, some 30mins driving north of Östersund. I'm not sure if there are more such deviations, even if I know only of this one from the top of my head.

There is a suggestion to pull the data from the Tesla API instead in this thread: https://github.com/adriankumpf/teslamate/discussions/1831
